gestalt-module/src/main/java/org/terasology/module/DependencyResolver.java

```diff
@@ -68,7 +68,17 @@ public DependencyResolver(ModuleRegistry registry) {
      * @return A set of compatible modules based on the required modules.
      */
     public ResolutionResult resolve(Name rootModule, Name... additionalModules) {
-        return resolve(Varargs.combineToSet(rootModule, additionalModules));
+        return resolve(true, rootModule, additionalModules);
+    }
+
+    /**
+     * @param includeOptional   If optional modules should be included
```

Comment:
What circumstances would a called include or not include optional modules?  How do they make that determination? Why not just have optional modules are included if possible?